From 732118aec356172189b17ef562e7e2072d610bb8 Mon Sep 17 00:00:00 2001
Message-ID: <732118aec356172189b17ef562e7e2072d610bb8.1773161960.git.teddy.astie@vates.tech>
In-Reply-To: <b8078ed45d66b94467b594ac7e8d849af32a0984.1773161960.git.teddy.astie@vates.tech>
References: <b8078ed45d66b94467b594ac7e8d849af32a0984.1773161960.git.teddy.astie@vates.tech>
From: =?UTF-8?q?Edwin=20T=C3=B6r=C3=B6k?= <edwin.torok@cloud.com>
Date: Fri, 25 Jul 2025 16:06:39 +0100
Subject: [PATCH 02/15] arch-x86/pmu.h: convert ascii art drawing to Unicode
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Use `aa2u` (ascii-art-to-unicode from Hackage) to convert to
Unicode box drawing characters.

The full list of supported drawing characters can be seen in the
examples at:
https://github.com/fmthoma/ascii-art-to-unicode/blob/master/src/Text/AsciiArt.hs

For future maintenance: conversion can be done incrementally
(mixing ascii art with already converted Unicode,
 and running the conversion tool again), or by hand.

No functional change.

Signed-off-by: Edwin Török <edwin.torok@cloud.com>
---
 xen/include/public/arch-x86/pmu.h | 120 +++++++++++++++---------------
 1 file changed, 60 insertions(+), 60 deletions(-)

diff --git a/xen/include/public/arch-x86/pmu.h b/xen/include/public/arch-x86/pmu.h
index 8be71a88ee..5bd0aa6f77 100644
--- a/xen/include/public/arch-x86/pmu.h
+++ b/xen/include/public/arch-x86/pmu.h
@@ -130,66 +130,66 @@ typedef struct xen_pmu_arch xen_pmu_arch_t;
 DEFINE_XEN_GUEST_HANDLE(xen_pmu_arch_t);
 
 /* Memory layout:
-*                .---------------------.
-*                | struct xen_pmu_data |
-* +==============+=====================+=======================+ <.
-* | vcpu_id                                                    |  |
-* +------------------------------------------------------------+  |
-* | pcpu_id                                                    |  |
-* +------------------------------------------------------------+  |
-* | domain_id                                                  |  |
-* +------------------------------------------------------------+  |
-* |##pad#######################################################|  |
-* +====+=+===+==================+==============================+  |
-* | pmu| | r | regs             |##pad#########################|  |
-* +----. +---. (xen or guest)   |##############################|  |
-* |      +======================+==============================+  |
-* |      | pmu_flags                                           |  |
-* |      +===+====================+============================+  |
-* |      | l | lapic_lvtpc        |############################|  |
-* |      +---. ###################|##pad#######################|  |
-* |      |     ###################|############################|  |
-* |      +===+=+=======+=====+====+====+=======+========+======+  |
-* |      | c | |       | amd |    |    | intel |         |#####|  |
-* |      +---+ |       .-----.    |    .-------.         |#####|  |
-* |      |     | counter          | fixed_counters       |#####|  | 
-* |      |     +------------------+----------------------+#####|  |
-* |      |     | ctrls            | arch_counters        |#####|  |
-* |      |     +=====+========+===+----------------------+#####|  |
-* |      |     |     | regs[] |  :| global_ctrl          |#####|  |
-* |      |     |     +--------.  :+----------------------+#####|  |
-* |      |     |struct           :| global_ovf_ctrl      |#####|  |
-* |      |     |xen_pmu_cntr_pair:+----------------------+#####|  |
-* |      |     |[counters]       :| global_status        |#####|  |
-* |      |     |                 :+----------------------+#####|  |
-* |      |     |                 :| fixed_ctrl           |#####|  |
-* |      |     |                 :+----------------------+#####|  |
-* |      |     |                 :| ds_area              |#####|  |
-* |      |     |                 :+----------------------+#####|  |
-* |      |     |                 :| pebs_enable          |#pad#|  |
-* |      |     |                 :+----------------------+#####|  |
-* |      |     |                 v| debugctl             |#####|  |
-* |      |     |##################+=======+========+=====+#####|  |
-* |      |     |##################|       | regs[] | :[0]|#####|  |
-* |      |     |##################|       +--------. :   |#####|  |
-* |      |     |##################| uint64_t         :   |#####|  |
-* |      |     |##################| [fixed_counters] :   |#####|  |
-* |      |     |##################|                  :   |#####|  |
-* |      |     |##################|                  :   |#####|  |
-* |      |     |##################| -----------------:   |#####|  |
-* |      |     |##################|  struct          :   |#####|  |
-* |      |     |##################| xen_pmu_cntr_pair:   |#####|  |
-* |      |     +==================+ [arch_counters]  :   +=====+  | 
-* |      |                        |                  :   |     |  |
-* |      |                        |                  v   |     |  |
-* |      |                        +======================+     |  |
-* |      +=====================================================+  |
-* +==========================+=================================+  |
-* |############################################################|  |
-* ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::  :
-* ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::  :
-* |############################################################|  | PAGE_SIZE
-* +=========+==================================================+ <.
+*                ╭─────────────────────╮
+*                │ struct xen_pmu_data │
+* ╒══════════════╧═════════════════════╧═══════════════════════╕ ◁╮
+* │ vcpu_id                                                    │  │
+* ├────────────────────────────────────────────────────────────┤  │
+* │ pcpu_id                                                    │  │
+* ├────────────────────────────────────────────────────────────┤  │
+* │ domain_id                                                  │  │
+* ├────────────────────────────────────────────────────────────┤  │
+* │██pad███████████████████████████████████████████████████████│  │
+* ╞════╤═╤═══╤══════════════════╤══════════════════════════════╡  │
+* │ pmu│ │ r │ regs             │██pad█████████████████████████│  │
+* ├────╯ ├───╯ (xen or guest)   │██████████████████████████████│  │
+* │      ╞══════════════════════╧══════════════════════════════╡  │
+* │      │ pmu_flags                                           │  │
+* │      ╞═══╤════════════════════╤════════════════════════════╡  │
+* │      │ l │ lapic_lvtpc        │████████████████████████████│  │
+* │      ├───╯ ███████████████████│██pad███████████████████████│  │
+* │      │     ███████████████████│████████████████████████████│  │
+* │      ╞═══╤═╤═══════╤═════╤════╪════╤═══════╤═══════════════╡  │
+* │      │ c │ │       │ amd │    │    │ intel │         │█████│  │
+* │      ├───┘ │       ╰─────╯    │    ╰───────╯         │█████│  │
+* │      │     │ counter          │ fixed_counters       │█████│  │
+* │      │     ├──────────────────┼──────────────────────┤█████│  │
+* │      │     │ ctrls            │ arch_counters        │█████│  │
+* │      │     ╞═════╤════════╤═══├──────────────────────┤█████│  │
+* │      │     │     │ regs[] │  ┆│ global_ctrl          │█████│  │
+* │      │     │     └────────╯  ┆├──────────────────────┤█████│  │
+* │      │     │struct           ┆│ global_ovf_ctrl      │█████│  │
+* │      │     │xen_pmu_cntr_pair┆├──────────────────────┤█████│  │
+* │      │     │[counters]       ┆│ global_status        │█████│  │
+* │      │     │                 ┆├──────────────────────┤█████│  │
+* │      │     │                 ┆│ fixed_ctrl           │█████│  │
+* │      │     │                 ┆├──────────────────────┤█████│  │
+* │      │     │                 ┆│ ds_area              │█████│  │
+* │      │     │                 ┆├──────────────────────┤█████│  │
+* │      │     │                 ┆│ pebs_enable          │█pad█│  │
+* │      │     │                 ┆├──────────────────────┤█████│  │
+* │      │     │                 ▽│ debugctl             │█████│  │
+* │      │     │██████████████████╞═══════╤════════╤═════╡█████│  │
+* │      │     │██████████████████│       │ regs[] │ ┆[0]│█████│  │
+* │      │     │██████████████████│       └────────╯ ┆   │█████│  │
+* │      │     │██████████████████│ uint64_t         ┆   │█████│  │
+* │      │     │██████████████████│ [fixed_counters] ┆   │█████│  │
+* │      │     │██████████████████│                  ┆   │█████│  │
+* │      │     │██████████████████│                  ┆   │█████│  │
+* │      │     │██████████████████│ ─────────────────┆   │█████│  │
+* │      │     │██████████████████│  struct          ┆   │█████│  │
+* │      │     │██████████████████│ xen_pmu_cntr_pair┆   │█████│  │
+* │      │     ╘══════════════════╡ [arch_counters]  ┆   ╞═════╡  │
+* │      │                        │                  ┆   │     │  │
+* │      │                        │                  ▽   │     │  │
+* │      │                        ╘══════════════════════╛     │  │
+* │      ╘═════════════════════════════════════════════════════╡  │
+* ╞════════════════════════════════════════════════════════════╡  │
+* │████████████████████████████████████████████████████████████│  │
+* ┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆  ┆
+* ┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆┆  ┆
+* │████████████████████████████████████████████████████████████│  │ PAGE_SIZE
+* ╘════════════════════════════════════════════════════════════╛ ◁╯
 */
 
 #endif /* __XEN_PUBLIC_ARCH_X86_PMU_H__ */
-- 
2.53.0

